cppForSwig/BtcUtils.h

```diff
@@ -881,6 +967,26 @@ class BtcUtils
                brr.advance( TxOutCalcLength(brr.getCurrPtr()) );
          }
       }
+
+      // Now extract the witnesses
+      if(usesWitness)
+      {
+         if(offsetsWitness != NULL)
+         {
+            offsetsWitness->resize(nIn + 1);
+            for (uint32_t i = 0; i < nIn; i++) {
+               (*offsetsWitness)[i] = brr.getPosition();
+               brr.advance(TxWitnessCalcLength(brr.getCurrPtr(), brr.getSizeRemaining()));
+            }
+         }
```

Comment:
missed the last entry of offsetsWitness:

offsetsWitness[nIn] = brr,getPosition();